tests/integration/mixer/outboundtrafficpolicy/helper.go

```diff
@@ -345,6 +317,104 @@ func RunExternalRequestTest(expected map[string][]string, t *testing.T) {
 		})
 }
 
+func RunExternalRequestMetricsTest(prometheus prometheus.Instance, mode TrafficPolicy, expected map[string]MetricsResponse, t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+		client, dest := setupEcho(t, ctx, mode)
```

Comment:
Not a huge blocker, but can/should we share the echo setup for metrics + status code? I was envisioning something along the lines:
```
TestExternal() {
  setupTest
  sendRequests()
  newSubTest("status", checkStatus())
  newSubTest("metrics", checkMetrics())
```

This seems to fit the model of "Do X thing, then make sure that Y and Z work". In this case,  "Do external traffic, then make sure that it succeeds and reports telemetry" . Rather than "Do X thing, then make sure that Y works. Do X thing, then make sure the Z works".

Reply:
Let me play around with it, but that would make maintenance easier long term as the mixer integration tests will go away and we'll just have telemetry v2.

Reply:
I did have an issue when I was first implementing the changes; if I checked the prometheus metrics after all of the test cases were run the metrics would quickly expire and the tests would fail. That's why I check after a test case has succeeded, but before it runs the next test case; see https://github.com/istio/istio/pull/22021/files#diff-d0f41b0f448a5437f788dbbfdb1ccd5eR353.

So I ran the integration tests with `--istio.test.nocleanup` and port-forward prometheus.  With this setup I would see the metrics be removed every ~5m or so, but I couldn't tell if they were cleaned up before each test. So there may be some prometheus tweaking that would need to be done to make sure it keeps things around until the test is finished. I'll update this when I have the changes implemented.